### DISCUSSION TITLE
MoltenVK and iOS

### DISCUSSION THREAD

**Discussion by egorov-k** [amerkoleci/Vortice.Vulkan#36] [General]
How to use Vortice.Vulkan on iOS?

**Comment by amerkoleci** [comment]
It already works as it loads moltenvk dylibs

**Reply by jurganson** [reply]
It is true that it loads the .dylib's from the VulkanSDK but .dylib are not allowed to be bundled and uploaded to the AppStore:
https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle

I have been struggling to get this to work by wrapping the .dylib in all manners of different configurations but having it in Versions/1.1/libMoltenVK.dylib OR in the root of the framework, in the root of the bundle, or anywhere results in Apple rejecting the bundle upload.

Do you have any tips or tricks for embedding the .dylib correctly such that Apple will not reject my upload to the AppStore? I am considering switching to a interop implementation of the vulkan bindings because right now I only have luck statically linking the libMoltenVK.a files from the SDK, but I would love to use Vortice in production.

**Reply by anose001** [reply]
@jurganson, Did you resolve this issue eventually? I am planning my project and if possible I would like to avoid being denied upload uf my app due to the AppStore not allowing .dylib files in the bundle

**Reply by jurganson** [reply]
@anose001 Yes I simply used the prebuild MoltenVK frameworks from the MoltenVK GitHub. Then in `vkInitialize` I needed to specify the direct path so along the lines of "path-to-your-frameworks-folder-root/MoltenVK.framework/MoltenVK". 

However I would suggest using TestFlight for iOS before doing too much work, as I believe the verification process is largely automated when it comes to checking for these kinds of things.
